components/uaepass-authenticator/src/main/java/org/wso2/carbon/identity/uaepass/authenticator/UAEPassAuthenticator.java

```diff
@@ -0,0 +1,1109 @@
+
+
+    /**
+     * Returns the federated IdP component's friendly name.
+     *
+     * @return String
+     */
+    @Override
+    public String getFriendlyName() {
+        return UAEPassAuthenticatorConstants.UAE.FEDERATED_IDP_COMPONENT_FRIENDLY_NAME;
+    }
+
+
+    /**
+     * Returns the federated IdP component name.
+     *
+     * @return String
+     */
+    @Override
+    public String getName() {
+        return UAEPassAuthenticatorConstants.UAE.FEDERATED_IDP_COMPONENT_NAME;
+    }
+
+
+    /**
+     * Returns the claim dialect URL.
+     *
+     * @return String
+     */
+    @Override
+    public String getClaimDialectURI() {
+        return UAEPassAuthenticatorConstants.UAEPassRuntimeConstants.OIDC_DIALECT;
+    }
+
+
+    /**
+     * Returns an unique string to identify each request and response separetely.
+     *
+     * @param request
+     * @return String
+     */
+    @Override
+    public String getContextIdentifier(HttpServletRequest request) {
+
+        String state = request.getParameter(UAEPassAuthenticatorConstants.UAE.OAUTH2_PARAM_STATE);
+        if (StringUtils.isNotBlank(state)) {
+            return state.split(",")[0];
+        } else {
+            LOG.error("An unique identifier couldn't issue for both Request and Response. ContextIdentifier is NULL");
+            return null;
+        }
+    }
+
+
+    /**
+     * Returns the all user input fields of federated authenticator component.
+     *
+     * @return List<Property> federated authenticator properties
+     */
+    @Override
+    public List<Property> getConfigurationProperties() {
+
+        List<Property> configProperties = new ArrayList<>();
+        Property clientId = new Property();
+        clientId.setName(UAEPassAuthenticatorConstants.UAE.CLIENT_ID);
+        clientId.setDisplayName("Client Id");
+        clientId.setRequired(true);
+        clientId.setDescription("Enter OAuth2/OpenID Connect client identifier value");
+        clientId.setType(UAEPassAuthenticatorConstants.UAEPassPropertyConstants.TEXTBOX);
+        clientId.setDisplayOrder(1);
+        configProperties.add(clientId);
+
+        Property clientSecret = new Property();
+        clientSecret.setName(UAEPassAuthenticatorConstants.UAE.CLIENT_SECRET);
+        clientSecret.setDisplayName("Client Secret");
+        clientSecret.setRequired(true);
+        clientSecret.setDescription("Enter OAuth2/OpenID Connect client secret value");
+        clientSecret.setType(UAEPassAuthenticatorConstants.UAEPassPropertyConstants.TEXTBOX);
+        clientSecret.setDisplayOrder(2);
+        clientSecret.setConfidential(true);
+        configProperties.add(clientSecret);
+
+        Property callbackUrl = new Property();
+        callbackUrl.setDisplayName("Callback URL");
+        callbackUrl.setRequired(true);
+        callbackUrl.setName(UAEPassAuthenticatorConstants.UAE.CALLBACK_URL);
+        callbackUrl.setDescription("The callback URL used to partner identity provider credentials.");
```

Comment:
Keep the consistency in all the description sentences. 
eg: use full stop or not, Wording